### PR TITLE
Fix #8460: autodata directive does not display the detail of TypeVars

### LINKED ISSUE #8460: Report underlying type information for NewType and TypeVar
**Is your feature request related to a problem? Please describe.**
Autodoc for NewType and TypeVar report the name of the type and the developer doc-comments but doesn't report the type metadata. 

**Describe the solution you'd like**
For NewType I'd expect it to report the underlying type.
For example:
MyInt = NewType('MyInt', int) should report in the generated docs that the underlying type is int.

ForTypeVar there are multiple attributes. If there's bound type information, that should be reported. Additionally, it should report whether the type is invariant, covariant, or contravariant. As a bonus, cross-linking between invariant, covariant, or contravariant TypeVar declarations for the same underlying type.

**Describe alternatives you've considered**
The alternative would be  for the developer to manually document the type information they've already declared - which seems redundant since this should be easy to resolve.

**Additional context**
- See also discussion in #8415

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- The details are only shown when automodule directive is used.  This
allows to document them via other auto* directives.
- refs: #8460

### CODE REVIEW

File: tests/test_ext_autodoc.py [hunk 1]
```diff
@@ -1750,27 +1758,31 @@ def test_autodoc_TypeVar(app):
         '',
         "   alias of TypeVar('T1')",
         '',
+        '',
         '.. py:data:: T3',
         '   :module: target.typevar',
         '',
         '   T3',
         '',
         "   alias of TypeVar('T3', int, str)",
```
Comment (tk0miya) [inline tests/test_ext_autodoc.py:1767]: I'll make these arguments hyperlinked in the next step.

File: sphinx/ext/autodoc/__init__.py [hunk 2]
```diff
@@ -2049,7 +2071,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
 
 
-class AttributeDocumenter(DocstringStripSignatureMixin, ClassLevelDocumenter, NewTypeMixin):  # type: ignore  # NOQA
+class AttributeDocumenter(DocstringStripSignatureMixin, ClassLevelDocumenter, NewTypeMixin, TypeVarMixin):  # type: ignore  # NOQA
```
Comment (tk0miya) [inline sphinx/ext/autodoc/__init__.py]: The order should be changed to correct MRO.

File: tests/test_ext_autodoc.py (hunk 1 above)
Comment (jp-larose) [reply tests/test_ext_autodoc.py:1767]: I don't see anything for the `bound`  parameter for TypeVar.  It should be accessible via the `__bound__` attribute.  It's important information to display.

Comment (tk0miya) [reply tests/test_ext_autodoc.py:1767]: Could you file a new issue? Then we'll consider improving it.

**Pull request merged**